### PR TITLE
Add support to multi-currency to Payment Selection Create From:

### PR DESCRIPTION
**Pull request by yamelsenih**
This pull request allows convert to payment selection currency all documents that will be paid. Currently exists a parameter for Smart Browser called Currency but it is used as query criteria and for convert amount of document, the problem is that always is filtered from selected currency and when it is converted is based on the same currency.

Just is change the follows Smart Browsers:
- Payment Selection Create From Invoice
- Generate Payment Selection (From Invoice) 
- Payment Selection Create From Payroll Movement

### CODE REVIEW

**Comment by yamelsenih** [comment]
Hi, I add some cache for improve performance for this process on Create From Payroll Process

**Review by marcalwestf — APPROVED** [review]
- Created and completed a Purchase Invoice in Euro
- Paid with Pay Selection in two partial payments: one in EUR and one in USD which totalled the open amount.
- The result was that every partial payment calculated correctly the paid amount in the courrency.
- In the end, the invoice wad paid completely by paying with two different currencies.

Test passed!

**Comment by marcalwestf** [comment]
@yamelsenih in order to get two partial payments, we had to change the WHERE clause of the Smart Browser _Create from Invoice_ (AD_Browse_ID=50054). 

This will be another Issue/Pull Request and does not interfere with your changes in this PR.

File: base/src/org/compiere/model/MPaySelectionLine.java [hunk 1]
```diff
@@ -182,18 +190,32 @@ public void setOrder(int orderId, BigDecimal amtSource,
 		setAmtSource(amtSource);
 		setOpenAmt(openAmt);
 		setPayAmt (payAmt);
+		if(order.getC_ConversionType_ID() > 0) {
```
Comment (yamelsenih) [inline base/src/org/compiere/model/MPaySelectionLine.java:193]: Epale

**Review by yamelsenih — COMMENTED** [review]
hola

File: base/src/org/compiere/process/PSCreateFromHRMovement.java [hunk 2]
```diff
@@ -45,20 +52,27 @@ protected void prepare() {
...
+			String paymentRule = getSelectionAsString(key, "HRM_PaymentRule");
+			BigDecimal sourceAmount = getSelectionAsBigDecimal(key, "HRM_Amount");
+			BigDecimal convertedAmount = getSelectionAsBigDecimal(key, "HRM_ConvertedAmount");
+			MPaySelectionLine line = new MPaySelectionLine(paySelection, sequence.getAndAdd(10), paymentRule);
 			//	Add Order
-			line.setHRMovement(HR_Movement_ID, Amount);
+			X_HR_Movement payrollMovement = new X_HR_Movement(getCtx(), movementId, get_TrxName());
+			X_HR_Process payrollProcess = payrollProcessMap.get(payrollMovement.getHR_Process_ID());
+			if(!Optional.ofNullable(payrollProcess).isPresent()) {
+				payrollProcess = (X_HR_Process) payrollMovement.getHR_Process();
```
Comment (e-Evolution) [inline base/src/org/compiere/process/PSCreateFromHRMovement.java:68]: @yamelsenih use directly if and default value :

Optional.ofNullable(payrollProcess).ifPresent( payrollProcess => {
}).orElseGet(() =>  payrollMovement.getHR_Process())

**Pull request merged**